components/CaseStatus/AddCaseStatusForm/ReviewAddCaseStatusForm.tsx

```diff
@@ -41,18 +49,33 @@ const ReviewAddCaseStatusForm: React.FC<{
     }
   };
 
+  const diplayObj: any = {
+    Type: [valueMapping[formAnswers.type as keyof CaseStatusMapping]],
+    'Start date': new Date(formAnswers.startDate).toLocaleDateString('en-GB', {
+      day: '2-digit',
+      month: 'short',
+      year: 'numeric',
+    }),
+    Notes: String(formAnswers.notes),
+  };
+
+  if (cpCaseStatusFields) {
+    cpCaseStatusFields.fields.map((elmField) => {
+      Object.keys(formAnswers).map((elm) => {
+        if (elm === elmField.name) {
+          elmField.options.map((option) => {
+            if (option.name === formAnswers[elm]) {
+              diplayObj[elmField.description] = option.description;
+            }
+          });
+        }
+      });
+    });
+  }
```

Comment:
Can you explain this please

Reply:
I think I could rename some of the variables to be clearer.

It loops through the form options retrieved from the API call to check if there are any matches in the the form answers and get the human readable description.

For example, if the user selected `CP` as the first option and then chose `Neglect` as a category of CP.

In the above case, the query sent to the review stage, has `CP` as the `type` parameter but `C1` as the category. The `Neglect` option the user sees is the description of the category selected i.e `C1`.

It might be easier to explain in person: The structure of our API request body to create a CP/LAC case status does not take the description `Neglect`, it takes the name of the option selected which is `C1`. This doesn't pass the work `Neglect` in the query parameters.

So the code above, loops through the options retrieved from the API, finds a match in the form answers, and if there is, rather than display the short hand code, returns the description so that we can display it to the user at the review stage.

Open to suggestions on better ways to do this.

The way this was coded was also in anticipation of LAC using the same process, so open to feedback on approach too.